server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -0,0 +1,144 @@
+import com.google.common.hash.Hashing
+import io.delta.kernel.Table
+import io.delta.kernel.defaults.engine.DefaultEngine
+import io.delta.kernel.engine.Engine
+import io.delta.kernel.internal.{ScanImpl, SnapshotImpl}
+import io.delta.standalone.DeltaLog
+import io.delta.standalone.internal.actions.{AddCDCFile, AddFile, Metadata, Protocol, RemoveFile}
+import io.delta.standalone.internal.exception.DeltaErrors
+import io.delta.standalone.internal.util.ConversionUtils
+import org.apache.commons.codec.digest.DigestUtils
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.fs.azure.NativeAzureFileSystem
+import org.apache.hadoop.fs.azurebfs.AzureBlobFileSystem
+import org.apache.hadoop.fs.s3a.S3AFileSystem
+import org.apache.spark.sql.types.{DataType, MetadataBuilder, StructType}
+import scalapb.{GeneratedMessage, GeneratedMessageCompanion}
+
+import io.delta.sharing.server.{AbfsFileSigner, CausedBy, DeltaSharedTableProtocol, DeltaSharingIllegalArgumentException, DeltaSharingUnsupportedOperationException, ErrorStrings, GCSFileSigner, PreSignedUrl, QueryResult, S3FileSigner, WasbFileSigner}
+import io.delta.sharing.server.config.TableConfig
+import io.delta.sharing.server.protocol.{QueryTablePageToken, RefreshToken}
+import io.delta.sharing.server.util.JsonUtils
+
+
+/**
+ * A util class stores all query parameters. Used to compute the checksum in the page token for
+ * query validation.
+ */
+private case class QueryParamChecksum(
+                                       version: Option[Long],
+                                       timestamp: Option[String],
+                                       startingVersion: Option[Long],
+                                       startingTimestamp: Option[String],
+                                       endingVersion: Option[Long],
+                                       endingTimestamp: Option[String],
+                                       predicateHints: Seq[String],
+                                       jsonPredicateHints: Option[String],
+                                       limitHint: Option[Long],
+                                       includeHistoricalMetadata: Option[Boolean])
+
+
+/**
+ * A table class that wraps `DeltaLog` to provide the methods used by the server.
+ */
+class DeltaSharedTableKernel(
+                        tableConfig: TableConfig,
+                        preSignedUrlTimeoutSeconds: Long,
+                        evaluatePredicateHints: Boolean,
+                        evaluateJsonPredicateHints: Boolean,
+                        evaluateJsonPredicateHintsV2: Boolean,
+                        queryTablePageSizeLimit: Int,
+                        queryTablePageTokenTtlMs: Int,
+                        refreshTokenTtlMs: Int) extends DeltaSharedTableProtocol {
+
+  protected val tablePath: Path = new Path(tableConfig.getLocation)
+
+
+  // Get the table and table client (engine). If shouldParallelLoadDeltaLog is true,
```

Comment:
nit: update the comment, shouldParallelLoadDeltaLog is not available, remove related content.

Reply:
fixed